src/server/middleware/errorHandler.js

```diff
@@ -1,9 +1,14 @@
-const { error } = require("winston");
+const { logger } = require("../services/winston");
 const PropsGenerator = require("../propsGenerator");
 
 const errorHandler = (err, req, res, next) => {
-  error(err);
-  error(`statusCode: ${res ? res.statusCode : err ? err.statusCode : null}`);
+  logger.error(err.message);
+  if (res.headersSent) {
+    return next(err);
+  }
+  logger.error(
+    `statusCode: ${res ? res.statusCode : err ? err.statusCode : null}`
+  );
   var props = {
     statusCode: res ? res.statusCode : err ? err.statusCode : "500",
     err: err ? err : "An error occurred.",
```

Comment:
If PropsGenerator is the cause of the error, line 15 will likely error again (this time uncaught). I think only call propsgenerator here if err doesn't contain the text "propsGenerator".

Reply:
might as well add quick unit test - I cna see ahving to mess with error handler again and breaking it!